tracer/src/Datadog.Trace/Logging/TracerFlare/TracerFlareManager.cs

```diff
@@ -0,0 +1,320 @@
+﻿// <copyright file="TracerFlareManager.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Datadog.Trace.Agent.DiscoveryService;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.RemoteConfigurationManagement;
+using Datadog.Trace.Vendors.Newtonsoft.Json;
+using Datadog.Trace.Vendors.Newtonsoft.Json.Linq;
+
+namespace Datadog.Trace.Logging.TracerFlare;
+
+internal class TracerFlareManager : ITracerFlareManager
+{
+    internal const string TracerFlareInitializationLog = "Enabling debug mode due to tracer flare initialization";
+    internal const string TracerFlareCompleteLog = "Disabled debug mode due to tracer flare complete";
+    internal const string ReceivedTracerFlareRequestLog = "Received tracer flare request";
+    private const int RevertGlobalDebugMinutes = 20;
+    private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor<TracerFlareManager>();
+
+    private readonly IDiscoveryService _discoveryService;
+    private readonly IRcmSubscriptionManager _subscriptionManager;
+    private readonly TracerFlareApi _flareApi;
+    private ISubscription? _subscription;
+    private Timer? _resetTimer = null;
+
+    public TracerFlareManager(
+        IDiscoveryService discoveryService,
+        IRcmSubscriptionManager subscriptionManager,
+        TracerFlareApi flareApi)
+    {
+        _subscriptionManager = subscriptionManager;
+        _flareApi = flareApi;
+        _discoveryService = discoveryService;
+    }
+
+    public bool? CanSendTracerFlare { get; private set; } = null;
+
+    public void Start()
+    {
+        _discoveryService.SubscribeToChanges(HandleConfigUpdate);
```

Comment:
Shouldn't this be inside the Interlocked.Exchange block as well?

Reply:
I... don't... know...